### ISSUE TITLE
main.html, Window "show" option and the background process

### ISSUE THREAD

**Issue by wehowski** [jyapayne/Web2Executable#151]
Hello there,
I have two different issues I would like to discribe here, maybe someone can give me a hint?

1.
As main html file I have one index.html.
In the windows options I disabled the option "show", so the index.html NEVER should appear.
However it DOES.

This is the part from my background script (inject-js start) where I open the window(s):
````
var Launch = function(){
			
     if(1 < Used.gui.App.argv.length || (uDIR === Used.path.basename( getAppPath() ) )){
          chrome.app.window.create(/* 'cli.html'*/
	   'http://app.frdl.webfan.de/cdn/application/webfan/cli.html', {
             id: 'cli.html',
           bounds: { width: 320, height: 320 }
         });
     }else{
     	     chrome.app.window.create('app.html', {
               id: 'app.html',
             bounds: { width: 320, height: 320 }
         });		
      	
      	 setTimeout(function(){   
      	    MainWinMenu();
     	    RegisterShortCuts();
     	 },1);        	
	}
};
 	
````
As you can see, if the app is started with more than one arguments, it should open cli.html, otherwise it should open app.html.
If the application is in the app modus, it accidently opens index.html, why?
So far, I believe it could have something to do with MainWinMenu(); /    RegisterShortCuts();
where I build the nw.gui.menu and try to register some shortcuts?
But I am not really sure how to stop it (it worked one time before)
 
2. As you can see above, the 'cli.html' I commented out and pu the location to an online file.
The reason is, localy the cli application does not work as expected.
If I load the online version, the cli app works as expected, but it is loading the background script wich throws 'process not defined'.
Maybe be my second issue is out of the scope of Web2Executable, but I am asking maybe one can give me a hint...?

Thanks.

**Comment by wehowski** [comment]
Update:
I wrapped a call to window.addEventListener in the background script into a process.nextTick function.
If I am correct, this resolved the first issue!

**Comment by jyapayne** [comment]
@wehowski Did you solve your first issue?
As for the second issue, it's hard to say what's wrong without knowing what your cli app does. But it sounds like you try to use a Node.js function from it. Do you use Node.js in the app.html file as well?

**Comment by wehowski** [comment]
> Did you solve your first issue?
Yes, I think so. 
One time the problem appeared again, but not now in my current version. I guess the problem comes from calling window. functions or nw.  get window functions or similar when the app.html is not open (e.g. the if app needs long to start because of low CPU), but I am not sure.
I did not test it without calling a crome.window.create, but I will do it soon, I expect that then the index.html will open.

(I'm not quite sure how to kill the process if no window is open, I will try chrome.runtime.onSuspend.addListener and/or chrome.idle.onStateChanged.addListener.
The app must not open a window in cli mode, but  I open it to use dev-tools to debug)

>Do you use Node.js in the app.html file as well?
It seems that node.js is working in app.html.
Node remote works as expected in my update script (e.g. requires fs from remote)
(I set the node remote option to my server domain, so the files SHOULD work online AND offline, but it does not with cli.html)
My cli script produces a function (successfully), but wich is not called correctly when using the local cli.html (I think it/the function hangs but could not debug it yet).
Online it works as expected, except the backgrund node.js as described above.

...I will try to provide a better test function for that, ...
If you like, you can test it to, everything can be found at 
/cdn/application/webfan/app.html
/cdn/application/webfan/cli.html
(The server I linked in the first post of the issue)
Node.js scripts/modules should work in ANY browser as I use a window.require function (with a fallback to global.window.top.__nw_require), except some core/system modules.

Maybe the problem is an issue with my require function behaves different in the app than in the browser!?
I will remove the core modules functions from the cli to test it next...

**Comment by jyapayne** [comment]
Again, this is an NW.js issue mainly. I wish I could be of more assistance, but I don't know your code well enough and it's very hard to follow. I also don't have the time to try to understand it. I suggest finding a minimum example that reproduces your behavior and filing a similar issue on the NW.js repo. Good luck!
